inductiva/utils/format_utils.py

```diff
@@ -40,55 +39,49 @@ def seconds_formatter(secs: float) -> str:
     return str(datetime.timedelta(seconds=round(secs)))
 
 
+#clean this function
+def apply_formatters(rows, columns, formatters):
+    data = {}
+
+    for index, column_name in enumerate(columns):
+        data[column_name] = [row[index] for row in rows]
+
+    for column_name, formatter in formatters.items():
+        if column_name in data:
+            data[column_name] = [formatter(x) for x in data[column_name]]
+
+    return data
```

Comment:
make this method simpler and delegate the dictionary construction to the caller.
Missing docstring and type hints.
See my other comment about this.

 ```suggestion
def apply_formatters(table: dict[str, Any], formatters: dict[str, Callable]):
    for column, formatter in formatters.items():
        if column in table:
            table[column] = [formatter(x) for x in table[column]]

    return table
```

perhaps we can add an option to apply the formatter inplace or over a copy of the input data:

```suggestion
def apply_formatters(table: dict[str, Any], formatters: dict[str, Callable], inplace: bool=False):
    table = table.copy() if inplace else table
    ...
    return table
```